scheduler_test.go

```diff
@@ -1058,3 +1058,31 @@ func TestScheduler_TagsUnique(t *testing.T) {
 	assert.EqualError(t, err, ErrTagsUnique(bar).Error())
 
 }
+
+func TestScheduler_LessParamsThanExpected(t *testing.T) {
+
+	s := NewScheduler(time.UTC)
+
+	f := func(s1, s2 string) {
+		fmt.Println("ok")
+	}
+
+	p := []interface{}{"p1"}
+	_, err := s.Every(1).Days().StartAt(time.Now().UTC().Add(time.Second*10)).Do(f, p...)
+	assert.EqualError(t, err, ErrWrongParams.Error())
+
+}
+
+func TestScheduler_MoreParamsThanExpected(t *testing.T) {
+
+	s := NewScheduler(time.UTC)
+
+	f := func(s1, s2 string) {
+		fmt.Println("ok")
+	}
+
+	p := []interface{}{"p1", "p2", "p3"}
+	_, err := s.Every(1).Days().StartAt(time.Now().UTC().Add(time.Second*10)).Do(f, p...)
+	assert.EqualError(t, err, ErrWrongParams.Error())
+
+}
```

Comment:
could you make the test a table test, and change the name to be for the `Do` func as i have below in the suggestion?

```suggestion
func TestScheduler_DoParameterValidation(t *testing.T) {

	testCases := []struct {
		description string
		parameters  []interface{}
	}{
		{"less than expected", []interface{}{"p1"}},
		{"more than expected", []interface{}{"p1", "p2", "p3"}},
	}
	
	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			s := NewScheduler(time.UTC)

			f := func(s1, s2 string) {
				fmt.Println("ok")
			}
			
			_, err := s.Every(1).Days().StartAt(time.Now().UTC().Add(time.Second*10)).Do(f, tc.parameters...)
			assert.EqualError(t, err, ErrWrongParams.Error())
		})
	}
}
```

Reply:
Hey @JohnRoesler 
Thanks for the suggestion, just push it